tasks/clientlibs.js

```diff
@@ -254,16 +259,18 @@ module.exports = function (grunt) {
 							clientLibJS = getIncludeString(clientLibName, 'js');
 
 							// Sort the array into dependency order
-							clientLibJSObj = performSort(clientLibJSObj);
+							clientLibJSOrderObj = performSort(clientLibJSObj);
 
 							// Generate and minify the string
-							for (var j = 0; j < clientLibJSObj.length; j = j + 1) {
-								clientLibJS += grunt.file.read(clientLibJSObj[j].fileName, { encoding: 'utf8' }) + '\r\n';
+							clientLibIncludeOrderObjJS = [];
+							for (var j = 0; j < clientLibJSOrderObj.length; j = j + 1) {
```

Comment:
Probably create the variable for "clientLibJSObj"/"clientLibJSOrderObj" length so you're not having to call the length property each time?

Reply:
You can see, I just made the change for switching object which have correct order. But good suggestion, I can update the code with your suggestion.